tests/test_modules.c

```diff
@@ -1270,6 +1272,194 @@ test_startup_data_foreign_identityref(void **state)
+    user = pwd->pw_name;
+
+    /* get group */
+    grp = getgrgid(getgid());
+    group = grp->gr_name;
+
+    /* params error, module_name NULL or owner NULL/group NULL/(int)perm=-1  or connection NULL */
+    sr_conn_ctx_t *conn = NULL;
+    ret = sr_set_module_access(conn, "test", user, group, 00666);
+    assert_int_equal(ret, SR_ERR_INVAL_ARG);
+
+    ret = sr_set_module_access(st->conn, NULL, user, group, 00666);
+    assert_int_equal(ret, SR_ERR_INVAL_ARG);
+
+    ret = sr_set_module_access(st->conn, "test", NULL, NULL, (mode_t)-1);
+    assert_int_equal(ret, SR_ERR_INVAL_ARG);
+
+    /* param perm error,invalid permissions */
+    ret = sr_set_module_access(st->conn, "test", user, group, 01777);
+    assert_int_equal(ret, SR_ERR_INVAL_ARG);
+
+    /* param perm error,setting execute permissions has no effect */
+    ret = sr_set_module_access(st->conn, "test", user, group, 00771);
+    assert_int_equal(ret, SR_ERR_INVAL_ARG);
+
+    /* non-existing module */
+    ret = sr_set_module_access(st->conn, "no-module", user, group, 00666);
+    assert_int_equal(ret, SR_ERR_NOT_FOUND);
+
+    /* invalid user */
+    ret = sr_set_module_access(st->conn, "test", "no-user", group, 00666);
+    assert_int_equal(ret, SR_ERR_NOT_FOUND);
+
+    /* invalid group */
+    ret = sr_set_module_access(st->conn, "test", user, "no-group", 00666);
+    assert_int_equal(ret, SR_ERR_NOT_FOUND);
+
+    /* user NULL and group NULL */
+    ret = sr_set_module_access(st->conn, "test", NULL, NULL, 00666);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    ret = sr_set_module_access(st->conn, "test", user, group, 00666);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* cleanup */
+    ret = sr_remove_module(st->conn, "test");
+    assert_int_equal(ret, SR_ERR_OK);
+}
+
+static void
+test_get_module_access(void **state)
+{
+    struct state *st = (struct state *)*state;
+    struct passwd *pwd;
+    struct group *grp;
+    const char *user;
+    char *owner, *group;
+    uint32_t conn_count;
+    int ret;
+    mode_t perm;
+
+    /* install module test */
+    ret = sr_install_module(st->conn, TESTS_DIR "/files/test.yang", TESTS_DIR "/files", NULL, 0);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* apply scheduled changes */
+    sr_disconnect(st->conn);
+    st->conn = NULL;
+    ret = sr_connection_count(&conn_count);
+    assert_int_equal(ret, SR_ERR_OK);
+    assert_int_equal(conn_count, 0);
+    ret = sr_connect(0, &st->conn);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* get user */
+    pwd = getpwuid(getuid());
+    user = pwd->pw_name;
+    /* get group */
+    grp = getgrgid(getgid());
+    group = grp->gr_name;
+
+    /* change module test permissions */
+    ret = sr_set_module_access(st->conn, "test", user, group, 00600);
+    assert_int_equal(ret, SR_ERR_OK);
+
+    /* params error, connection NULL or module name NULL or ower/group/perm NULL */
+    sr_conn_ctx_t *conn = NULL;
+    ret = sr_get_module_access(conn, "test", &owner, &group, &perm);
```

Comment:
Seems `conn` variable is redundant, you can simply use `NULL` in the test.

Reply:
ok, I have modified it.